controllers/component/component_adapter.go

```diff
@@ -105,12 +106,24 @@ func (a *Adapter) EnsureComponentIsCleanedUp() (controller.OperationResult, erro
 			return controller.RequeueWithError(err)
 		}
 	}
+	a.component, err = a.loader.GetComponent(a.client, a.context, a.component.Name, a.component.Namespace)
```

Comment:
Maybe add a comment before, that we are doing it to refresh state of component to minimize race condition with updating finalizers

Reply:
Agreed.